### PR TITLE
[ABA] Parallel BA request + naive formula + warning messages

### PR DESCRIPTION
**Pull request by W3stside**
# Summary

This runs the BA trade in parallel. Faster than previous logic.

Uses the naive formula: `(IV - FV) / IV / 2 * 100` via inverse trading:
1. initial trade was a SELL? Simulate a BUY order
2. Initial trade was a BUY? Simulate a SELL order
e.g
```
SELL: 100 DAI for 5 ETH
// so ABA in this case:
BUY: 100 DAI for 6 ETH
// ---------------------
(5- 6) / 5 / 2 * 100 = -10%
```

  # To Test

1. select tokens
2. trade

### CODE REVIEW

**Comment by elena-zh** [comment]
Text color in warnings is different
image
image


And still I can't press on the Swap button when no warnings are displayed on the form 
image

File: src/custom/hooks/usePriceImpact/useFallbackPriceImpact.ts [hunk 1]
```diff
@@ -109,26 +77,21 @@ export default function useFallbackPriceImpact({ abTrade, fiatPriceImpact }: Fal
     if (quoteError) {
       setImpact(undefined)
       setError(quoteError)
-    } else if (!loading && abIn && abOut && baOut) {
-      const impact = calculateFallbackPriceImpact(isExactIn ? abIn : abOut, baOut)
+    } else if (!loading && abIn && abOut && baIn && baOut) {
+      let impact = undefined
+      if (isExactIn) {
+        impact = calculateFallbackPriceImpact(abOut, baIn)
```
[the remark this answers is not in the document]
Comment (anxolin) [reply src/custom/hooks/usePriceImpact/useFallbackPriceImpact.ts:83]: Now i get it, kind of, but maybe we can add a few comments to make it easier. 
reasoning why abOut and baIn is the right values here is hard

**Review by anxolin — COMMENTED** [review]
Nice, but I have come concerns or comments

## Simplify vs Complex
I like the simple model, it could work! If we see the numbers are good, we can keep it! 😃

## Error with calculation
As explained in slack, there's cases where values are wrong


https://gnosisinc.slack.com/archives/CSGLUC1BN/p1638445466212400?thread_ts=1638437242.207100&cid=CSGLUC1BN

In the thread we see how we induce the user to think he is making `87%` but they loose half, so it should near `-50%`

## Current strategy
You used Felix/Nick approach, which I believe has PROS/CONS

The PROs are:
* You can start to calculate the BA trade earlier. How early? as soon as the user type something in the INPUT

The CONS are:
* Is less intuitive, not a big one, but there will be slight differences in the calculation, and the mental model of a calculating this for a user is, I you buy something, and then you sell it, how much you get back?
* More important problem for me is, BUY orders can only buy a finite amount, SELL orders can sell infinite amounts. For this reasons, in situations of low liquidity using the SELL order for the (BA trade) will be superior.

## Summary
My biggest problem is that, this PR uses the "optimization" approach, but it doesn't use the optimisation (the function receives a fully calculated AB trade already). So we have the CONS and not the PROS

Also, i think the SELL for BA trade is better for measuring illiquid markets

**Comment by anxolin** [comment]
To explain the options I see and PROS/CONS i have done this small schema:

IMG_3559

**Comment by anxolin** [comment]
Adding my comment here:
https://gnosisinc.slack.com/archives/CSGLUC1BN/p1638437554208000?thread_ts=1638437242.207100&cid=CSGLUC1BN

Can we soften up the tooltip message? I like more something like:
```
We can't calculate the price impact.

You can still move forward, but __please review carefully that the receive amounts are what your expect.__
```

**Comment by W3stside** [comment]
thanks for all the comments, closing in favour of #1960

**Pull request closed without merging**